examples/keyvalue-inmemory/link.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"github.com/wasmCloud/provider-sdk-go"
+	"log"
+)
+
+func (p *Provider) establishSourceLink(link provider.InterfaceLinkDefinition) error {
+	if _, exists := p.sourceLinks[link.Target]; exists {
+		log.Println("Source link already exists, ignoring duplicate", link)
+		return nil
+	}
```

Comment:
Nothing wrong with having this check, but inside of the provider SDK we do handle this duplication check and avoid delivering the link at all, so I think we can remove it https://github.com/wasmCloud/provider-sdk-go/blob/main/provider.go#L340

Reply:
Sounds good! I can remove this.